components/post_view/combined_system_message/combined_system_message.jsx

```diff
@@ -0,0 +1,284 @@
+        one: {
+            id: 'combined_system_message.left_team.one',
+            defaultMessage: '{firstUser} <b>left the team</b>.',
+        },
+        two: {
+            id: 'combined_system_message.left_team.two',
+            defaultMessage: '{firstUser} and {secondUser} <b>left the team</b>.',
+        },
+        many_expanded: {
+            id: 'combined_system_message.left_team.many_expanded',
+            defaultMessage: '{users} and {lastUser} <b>left the team</b>.',
+        },
+    },
+};
+
+export default class CombinedSystemMessage extends React.PureComponent {
+    static propTypes = {
+        currentUserId: PropTypes.string.isRequired,
+        messageData: PropTypes.array.isRequired,
+        allUserIds: PropTypes.array.isRequired,
+        actions: PropTypes.shape({
+            getMissingProfilesByIds: PropTypes.func.isRequired,
+        }).isRequired,
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            missingProfiles: [],
+        };
+    }
+
+    componentDidMount() {
+        this.loadMissingProfiles();
+    }
+
+    loadMissingProfiles = async () => {
+        const {
+            actions,
+            allUserIds,
+        } = this.props;
+        const missingProfiles = await actions.getMissingProfilesByIds(allUserIds);
+        this.setState({missingProfiles});
+    }
+
+    getDisplayNameByIds = (userIds = []) => {
+        const {currentUserId} = this.props;
+
+        const displayNames = userIds.
+            filter((userId) => {
+                return userId !== currentUserId;
+            }).
+            map((userId) => {
+                return getDisplayNameByUserId(userId, true);
+            });
+
+        const userInProp = userIds.includes(currentUserId);
+        if (userInProp) {
+            displayNames.unshift(localizeMessage('combined_system_message.you', 'You'));
+        }
+
+        return displayNames;
+    }
+
+    renderFormattedMessage(postType, userIds, actorId) {
+        const userDisplayNames = this.getDisplayNameByIds(userIds);
+        let actor = actorId ? this.getDisplayNameByIds([actorId])[0] : '';
+        if (actorId === this.props.currentUserId) {
+            actor = actor.toLowerCase();
+        }
+
+        const firstUser = userDisplayNames[0];
+        const numOthers = userDisplayNames.length - 1;
+
+        let formattedMessage;
+        if (numOthers === 0) {
+            formattedMessage = (
+                <FormattedHTMLMessage
+                    id={postTypeMessage[postType].one.id}
+                    defaultMessage={postTypeMessage[postType].one.defaultMessage}
+                    values={{
+                        firstUser,
+                        actor,
+                    }}
+                />
+            );
+
+            if (
+                userIds[0] === this.props.currentUserId &&
+                ([ADD_TO_CHANNEL, ADD_TO_TEAM, REMOVE_FROM_CHANNEL, REMOVE_FROM_TEAM].includes(postType))
```

Comment:
Instead of checking the postType, you could see if `postTypeMessage[postType].one_you` exists in case we add more messages like "You joined the channel"